packages/grafana-ui/src/types/datasource.ts

```diff
@@ -16,10 +17,9 @@ import { PluginMeta, GrafanaPlugin } from './plugin';
 import { PanelData } from './panel';
 import { Observable } from 'rxjs';
 
-// NOTE: this seems more general than just DataSource
-export interface DataSourcePluginOptionsEditorProps<TOptions> {
-  options: TOptions;
-  onOptionsChange: (options: TOptions) => void;
+export interface DataSourcePluginOptionsEditorProps<JSONData = DataSourceJsonData, SecureJSONData = {}> {
```

Comment:
Changed the type here as it seems like using anything else than `DataSourceSettings` would be a mistake anyway.

Reply:
As a note for future - this kind of changes(as well as the ones with changing ComponentClass to ComponentType) we should introduce in a separate PR not to blur the picture.

Reply:
Yeah this one was kinda orthogonal but `ComponentType` was needed otherwise this refactor would not work.